src/utils/cacheClientUtils.ts

```diff
@@ -112,3 +112,5 @@ export const createInMemoryCacheClientWithNamespace = <T>(namespace: string) =>
     );
 
 export const getResultsFromCache = async (key: string): Promise<CachedResult> => getCacheStrategyInstance().getValueFromCache(CacheNamespaces.RESULTS_NAMESPACE, key);
+
+export const refreshTtlForCachedResult = async <T>(key: string, ttl: number, value: CachedResult<T>): Promise<void> => getCacheStrategyInstance().refreshTtlForCachedResult(CacheNamespaces.RESULTS_NAMESPACE, key, ttl, value);
```

Comment:
import {scheduleRefreshTtlTask} from 'queueUtils.ts'

I suggest here a more common approach that can be used in every strategy to not redundant some parts of code.
1. Leave the refreshing part in strategies as it was before - without calling getRefreshTtlQueue() inside strategy. So logic there should do only the 'refreshing part'
2. In queueUtils register in Container a js Set() which will hold for us information for all already scheduled keys to refresh.  
3. In queueUtils implement a small wrapper, which will check if the given task is already scheduled. It should simply get that set from Container, and check it with method `.has()` if the given key was already scheduled to be refreshed
4.  Here in the cacheClientUtils firstly check if the task was scheduled, if not then push it to the queue. 
5. In queue worker execute given task, and after execution remove the key from the set from point number 2 - so task for it can be scheduled again.